src/modules/items/QuestItem.ts

```diff
@@ -2,11 +2,13 @@ import QuestLineState from '../quests/QuestLineState';
 import Item from './Item';
 
 export default class QuestItem extends Item {
-    constructor(name: string, displayName : string, description : string, private questlineName : string) {
+    constructor(name: string, displayName : string, description : string, private questlineName : string, private endQuestlineName : string = undefined) {
         super(name, undefined, undefined, undefined, displayName, description, 'quest');
+        endQuestlineName = endQuestlineName ?? questlineName;
```

Comment:
This trying to default as questlineName doesn't actually work, because you are setting a local variable instead of this.endQuestLineName.

I think you could also just do this in the param list:
```suggestion
    constructor(name: string, displayName : string, description : string, private questlineName : string, private endQuestlineName = questlineName) {
        super(name, undefined, undefined, undefined, displayName, description, 'quest');
```

Reply:
Looks fun. Will test it later